packages/studio-base/src/panels/Plot/PlotLegendRow.tsx

```diff
@@ -244,6 +263,9 @@ export default function PlotLegendRow({
         </div>
       )}
       <div className={classes.actions}>
+        <IconButton size="small" title="Edit settings" onClick={() => setSettingsModalOpen(true)}>
```

Comment:
I think you can solve the sizing issue by applying the same className as the `removeIcon`. I would consider renaming it to something like `actionIcon` though.

```suggestion
        <IconButton classes={classes.actionIcon} size="small" title="Edit settings" onClick={() => setSettingsModalOpen(true)}>
```